From 01488f61ff0219e323db99e4a2a10f806b7372f5 Mon Sep 17 00:00:00 2001
From: Chris Blake <chrisrblake93@gmail.com>
Date: Sat, 4 Apr 2020 10:28:38 -0500
Subject: [PATCH] Fix AP & Linux 5.6 Compile

---
 os_dep/linux/ioctl_cfg80211.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/os_dep/linux/ioctl_cfg80211.c b/os_dep/linux/ioctl_cfg80211.c
index b7ab5fd..0045957 100644
--- a/os_dep/linux/ioctl_cfg80211.c
+++ b/os_dep/linux/ioctl_cfg80211.c
@@ -348,7 +348,10 @@ rtw_cfg80211_default_mgmt_stypes[NUM_NL80211_IFTYPES] = {
 
 static u64 rtw_get_systime_us(void)
 {
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,20,0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5,6,0))
+	struct timespec64 ts = ktime_to_timespec64(ktime_get_boottime());
+	return ((u64)ts.tv_sec*1000000) + ts.tv_nsec / 1000;
+#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(4,20,0))
 	struct timespec ts = ktime_to_timespec(ktime_get_boottime());
 	return ((u64)ts.tv_sec*1000000) + ts.tv_nsec / 1000;
 #elif (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,39))
@@ -3617,6 +3620,9 @@ void rtw_cfg80211_indicate_sta_assoc(_adapter *padapter, u8 *pmgmt_frame, uint f
 			ie_offset = _REASOCREQ_IE_OFFSET_;
 	
 		sinfo.filled = 0;
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4, 18, 0))
+		sinfo.pertid = 0;
+#endif
 // cf commit 319090bf6c75e3ad42a8c
 //		sinfo.filled = STATION_INFO_ASSOC_REQ_IES;
 		sinfo.assoc_req_ies = pmgmt_frame + WLAN_HDR_A3_LEN + ie_offset;
-- 
2.17.1

