--- a/arch/arm64/kernel/cpuinfo.c
+++ b/arch/arm64/kernel/cpuinfo.c
@@ -165,6 +165,9 @@ static int c_show(struct seq_file *m, void *v)
 		seq_printf(m, "CPU part\t: 0x%03x\n", MIDR_PARTNUM(midr));
 		seq_printf(m, "CPU revision\t: %d\n\n", MIDR_REVISION(midr));
 	}
+	seq_printf(m, "Hardware\t: %s\n", "Allwinnersun50iw2Family");
+	seq_printf(m, "Revision\t: %s\n", "0000");
+	seq_printf(m, "Serial\t\t: %s\n", "0000000000000000");

 	return 0;
 }
